src/base.satyg

```diff
@@ -37,4 +37,8 @@ let finish = BaseSub.finish
 type 'a thunk = unit -> 'a
 
 %% [run-thunk t] evaluates thunk [t] and returns the result.
-let run-thunk t = t ()
\ No newline at end of file
+let run-thunk t = t ()
+
+type 'ok 'err result =
+| Ok of 'ok
+| Err of 'err
```

Comment:
Moving this here because I wanted to use the result type in a module result.satyg imports

Reply:
BTW, in version 2.0.0 I wish to have all types (e.g. Char.t, Stream.t) in satysfi-base here.
This is because satysfi does not allow cycles in dependency graphs of modules and for example if we import result.satyg in stream.satyg we cannot declare functions using Stream.t in result.satyg.